vault/expiration.go

```diff
@@ -841,11 +841,18 @@ type leaseEntry struct {
 }
 
 // encode is used to JSON encode the lease entry
-func (l *leaseEntry) encode() ([]byte, error) {
-	return json.Marshal(l)
+func (le *leaseEntry) encode() ([]byte, error) {
+	return json.Marshal(le)
 }
 
-func (le *leaseEntry) renewable() error {
+func (le *leaseEntry) renewable() bool {
+	if err := le.renewableErr(); err == nil {
+		return true
+	}
+	return false
+}
+
+func (le *leaseEntry) renewableErr() error {
```

Comment:
I'm concerned about people defaulting to use this function when in fact the exact error returned can be useful, e.g. in a debugging/logs context.